pkg/skaffold/instrumentation/meter.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2020 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package instrumentation
+
+import (
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner/runcontext"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/version"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/yaml"
+	"github.com/GoogleContainerTools/skaffold/proto"
+)
+
+type skaffoldMeter struct {
+	ExitCode       int
+	BuildArtifacts int
+	Command        string
+	Version        string
+	OS             string
+	Arch           string
+	PlatformType   string
+	Deployers      []string
+	EnumFlags      map[string]interface{}
+	Builders       map[string]bool
+	SyncType       map[string]bool
+	DevIterations  map[string]int
+	StartTime      time.Time
+	ErrorCode      proto.StatusCode
+}
+
+var (
+	meter = skaffoldMeter{
```

Comment:
Is there a reason why this is a `var` 
I am thinking, if we need to make this struct thread safe. e.g. 
- would you accept this struct to be updated in different go subroutines?
e.g. - in case a build fails, does the struct count get updated then and there ?
Look at pkg/skaffold/event/State struct and how its protected with a mutex.

Reply:
At the moment I believe everything that would be captured in this struct can be done outside of go subroutines. But I do not think it should exclude those, given that I think the const label is more appropriate. I did not really have any intent with picking var 😅